client/asset/ltc/ltc.go

```diff
@@ -158,6 +158,27 @@ func (d *Driver) Create(params *asset.CreateWalletParams) error {
 		params.Logger, recoveryCfg.NumExternalAddresses, recoveryCfg.NumInternalAddresses, chainParams)
 }
 
+// customSPVWalletConstructors are functions for setting up custom
+// implementations of the btc.BTCWallet interface that may be used by the
+// ExchangeWalletSPV instead of the default spv implementation.
+var customSPVWalletConstructors = map[string]btc.CustomSPVWalletConstructor{}
+
+// RegisterCustomSPVWallet registers a function that should be used in creating
+// a btc.BTCWallet implementation that the ExchangeWalletSPV will use in place
+// of the default spv wallet implementation. External consumers can use this
+// function to provide alternative btc.BTCWallet implementations, and must do so
+// before attempting to create an ExchangeWalletSPV instance of this type.
+func RegisterCustomSPVWallet(constructor btc.CustomSPVWalletConstructor, def *asset.WalletDefinition) error {
+	for _, availableWallets := range WalletInfo.AvailableWallets {
+		if def.Type == availableWallets.Type {
+			return fmt.Errorf("(%q): %w", def.Type, asset.ErrWalletTypeAlreadyRegistered)
+		}
+	}
+	customSPVWalletConstructors[def.Type] = constructor
+	WalletInfo.AvailableWallets = append(WalletInfo.AvailableWallets, def)
+	return nil
+}
+
```

Comment:
Why does **bch** use the `btc` `RegisterCustomSPVWallet` but **ltc** has it's own and duplicates the btc constructor code?

Reply:
RegisterCustomSPVWallet was not implemented for bch cuz Cryptopower only supports btc, ltc and dcr atm.